### ISSUE TITLE
Time out and unknown state

### ISSUE THREAD

**Issue by tbalthazar** [canonical/multipass#1268]
Hi,

I'm on Ubuntu 19.10, I tried to start 2 instances, got a time out message and they are now in an "unknown" state:
```
~$ multipass list
Name                    State             IPv4             Image
eaon2                   Stopped           --               Ubuntu 19.10
eoan                    Stopped           --               Ubuntu 19.10
ssh                     Stopped           --               Ubuntu 18.04 LTS
wg-home-server          Running           10.152.149.167   Ubuntu 19.10
wg-vps                  Running           10.152.149.212   Ubuntu 19.10
~$ multipass stop wg-vps wg-home-server
~$ multipass list
Name                    State             IPv4             Image
eaon2                   Stopped           --               Ubuntu 19.10
eoan                    Stopped           --               Ubuntu 19.10
ssh                     Stopped           --               Ubuntu 18.04 LTS
wg-home-server          Stopped           --               Ubuntu 19.10
wg-vps                  Stopped           --               Ubuntu 19.10
~$ multipass start wg-vps wg-home-server
start failed: The following errors occurred:                                    
wg-vps: timed out waiting for response
wg-home-server: timed out waiting for response
~$ multipass list
Name                    State             IPv4             Image
eaon2                   Stopped           --               Ubuntu 19.10
eoan                    Stopped           --               Ubuntu 19.10
ssh                     Stopped           --               Ubuntu 18.04 LTS
wg-home-server          Unknown           --               Ubuntu 19.10
wg-vps                  Unknown           --               Ubuntu 19.10
~$ multipass version
multipass  1.0.1
multipassd 1.0.1
~$ uname -a
Linux thinkpad 5.3.0-24-generic #26-Ubuntu SMP Thu Nov 14 01:33:18 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
```

Trying to start one of the VM in Unknown state:
```
~$ multipass start wg-vps
Starting wg-vps -[2019-12-23T17:19:37.371] [error] [wg-vps] process error occurred Crashed program: qemu-system-x86_64; error: Process crashed

 /[2019-12-23T17:19:37.373] [error] [wg-vps] error: program: qemu-system-x86_64; error: Process crashed

start failed: cannot connect to the multipass socket                            
Please ensure multipassd is running and '/var/snap/multipass/common/multipass_socket' is accessible
~$ multipass ls
list failed: cannot connect to the multipass socket
Please ensure multipassd is running and '/var/snap/multipass/common/multipass_socket' is accessible
~$ multipass ls
Name                    State             IPv4             Image
eaon2                   Stopped           --               Ubuntu 19.10
eoan                    Stopped           --               Ubuntu 19.10
ssh                     Stopped           --               Ubuntu 18.04 LTS
wg-home-server          Stopped           --               Ubuntu 19.10
wg-vps                  Stopped           --               Ubuntu 19.10

```

**Comment by ChrisDowning** [comment]
Seeing the same thing. When Multipass is freshly installed (on an Ubuntu 18.04 system - Multipass version 1.0.1), I can start a VM and connect to it fine. After suspendingrestarting the host machine, ```multipass list``` will first show the VM as being in the "Starting" state, but then fail to start and shows instead in either the "Suspended" or "Unknown" state.

When the VM is in "Unknown" state, attempting to start it returns the error seen above, ie:
```
Starting DevServer -[2019-12-27T22:37:00.445] [error] [DevServer] process error occurred Crashed program: qemu-system-x86_64; error: Process crashed

 /[2019-12-27T22:37:00.446] [error] [DevServer] error: program: qemu-system-x86_64; error: Process crashed

start failed: cannot connect to the multipass socket
Please ensure multipassd is running and '/var/snap/multipass/common/multipass_socket' is accessible
```

Restarting the snap daemon causes ```multipass list``` to correctly show the VM as "Suspended". Attempting to start from this position returns:
```
start failed: The following errors occurred:
DevServer: timed out waiting for response
```

So far, I've only managed to successfully use Multipass during the initial window between running ```sudo snap install multipass --classic``` and exiting the login session.

**Comment by townsend2010** [comment]
Hi @tbalthazar,

Looks like you're hitting a couple of issues here :frowning_face: Let's try to hit this one at a time.

First, for the timeout, could you do `journalctl -fu snap.multipass.*` in one terminal and then try the `start` command and then post the output here?  Hopefully that will shed some light on why the instances are not responding.

For the second issue, it looks like we are not handling the `UNKNOWN` state correctly when a `start` is issued on an instance in this state.  It look like the daemon is crashing.  I will try to debug this.

**Comment by ChrisDowning** [comment]
Hi @townsend2010 - I recently updated to 19.10 so decided to try again, but saw the same behaviour - here is the output of following your instructions:

Initial state (`multipass list`):
```
Name                    State             IPv4             Image
ubuntu1804              Suspended         --               Ubuntu 18.04 LTS
```
Output of `journalctl -fu snap.multipass.*` **before** running `multipass start`:
```-- Logs begin at Sat 2019-06-08 19:10:14 BST. --
Jan 08 23:13:55 XPS13 dnsmasq[12951]: compile time options: IPv6 GNU-getopt DBus i18n IDN DHCP DHCPv6 no-Lua TFTP conntrack ipset auth DNSSEC loop-detect inotify
Jan 08 23:13:55 XPS13 dnsmasq-dhcp[12951]: DHCP, IP range 10.106.123.2 -- 10.106.123.254, lease time infinite
Jan 08 23:13:55 XPS13 dnsmasq-dhcp[12951]: DHCP, sockets bound exclusively to interface mpqemubr0
Jan 08 23:13:55 XPS13 dnsmasq[12951]: using local addresses only for domain multipass
Jan 08 23:13:55 XPS13 dnsmasq[12951]: reading /etc/resolv.conf
Jan 08 23:13:55 XPS13 dnsmasq[12951]: using local addresses only for domain multipass
Jan 08 23:13:55 XPS13 dnsmasq[12951]: using nameserver 127.0.0.53#53
Jan 08 23:13:55 XPS13 dnsmasq[12951]: read /etc/hosts - 7 addresses
Jan 08 23:13:55 XPS13 dnsmasq[12951]: cannot read /var/snap/multipass/common/data/multipassd/network/dnsmasq.hosts: No such file or directory
Jan 08 23:13:55 XPS13 multipassd[12891]: gRPC listening on unix:/var/snap/multipass/common/multipass_socket, SSL:on
```

Output of `journalctl -fu snap.multipass.*` **after** running `multipass start`:
```
Jan 08 23:18:30 XPS13 multipassd[12891]: Cannot determine QEMU machine type. Defaulting to 'pc-i440fx-xenial'.
Jan 08 23:18:30 XPS13 multipassd[12891]: Loading AppArmor policy:
...(snip)...
Jan 08 23:18:30 XPS13 multipassd[12891]: process working dir '/snap/multipass/1577/qemu'
Jan 08 23:18:30 XPS13 multipassd[12891]: process program 'qemu-system-x86_64'
Jan 08 23:18:30 XPS13 multipassd[12891]: process arguments '--enable-kvm, -device, virtio-scsi-pci,id=scsi0, -drive, file=/var/snap/multipass/common/data/multipassd/vault/instances/ubuntu1804/ubuntu-18.04-server-cloudimg-amd64.img,if=none,format=qcow2,discard=unmap,id=hda, -device, scsi-hd,drive=hda,bus=scsi0.0, -smp, 1, -m, 1024M, -device, virtio-net-pci,netdev=hostnet0,id=net0,mac=52:54:00:2f:87:83, -netdev, tap,id=hostnet0,ifname=tap-f74f21ad1ac,script=no,downscript=no, -qmp, stdio, -cpu, host, -chardev, null,id=char0, -serial, chardev:char0, -nographic,
-cdrom, /var/snap/multipass/common/data/multipassd/vault/instances/ubuntu1804/cloud-init-config.iso, -loadvm, suspend, -machine, pc-i440fx-xenial'
Jan 08 23:18:30 XPS13 multipassd[12891]: Resuming from a suspended state
Jan 08 23:18:30 XPS13 multipassd[14124]: Applying AppArmor policy: multipass.ubuntu1804.qemu-system-x86_64
Jan 08 23:18:30 XPS13 multipassd[12891]: process started
Jan 08 23:18:30 XPS13 multipassd[12891]: QMP: {"QMP": {"version": {"qemu": {"micro": 1, "minor": 11, "major": 2}, "package": "(Debian 1:2.11+dfsg-1ubuntu7.21)"}, "capabilities": []}}
Jan 08 23:18:31 XPS13 multipassd[12891]: qemu-system-x86_64: warning: Unknown firmware file in legacy mode: etc/msr_feature_control
Jan 08 23:18:31 XPS13 multipassd[12891]:
Jan 08 23:18:31 XPS13 multipassd[12891]: qemu-system-x86_64: Machine type received is 'pc-i440fx-bionic' and local is 'pc-i440fx-xenial'
Jan 08 23:18:31 XPS13 multipassd[12891]:
Jan 08 23:18:31 XPS13 multipassd[12891]: qemu-system-x86_64: Error -22 while loading VM state
Jan 08 23:18:31 XPS13 multipassd[12891]:
Jan 08 23:18:31 XPS13 multipassd[12891]: QMP: {"return": {}}
```

Response from `multipass list`:
```
start failed: The following errors occurred:
ubuntu1804: timed out waiting for response
```
Hope that helps.

**Comment by townsend2010** [comment]
@ChrisDowning,

It looks you are running into a different problem than what is in the initial bug description.  You have suspended instance and it seems like a Multipass upgrade broke this :frowning_face:

Do you have important data on that instance?  You should be able to create a new instance with no problems and use that.  And if you don't have anything important on the existing instance, you can delete and purge it.

**Comment by tbalthazar** [comment]
Hi @townsend2010,

Thanks for getting back to me.
I upgraded to multipass 1.0.2 and I cannot reproduce the issue anymore.
I'll reopen this issue if the problem occurs again.

**Comment by ChrisDowning** [comment]
@townsend2010 Thanks - sorry for hijacking; this was the closest I found to the symptoms I was seeing. Like the OP, upgrading to 1.0.2 seems to have resolved the problem - I've been able to restart my host several times and existing VMs come back up without issue. Nothing to worry about with the broken VM, it was just a test and had no data/configuration.

**Comment by townsend2010** [comment]
@ChrisDowning, no worries at all.  The symptoms were the same for sure.  Glad it's working for you!

**Comment by YousefSaber** [comment]
well I'm facing the same issue despite using Multipass v1.0.2
https://github.com/canonical/multipass/issues/1358

**Comment by Ian-T-Price** [comment]
I experienced a very rare (once in five years) complete system hang while running two multipass VMs.

After a system restart both VMs were in an unknown state and no combination of multipass (ver 1.1.0) commands revived them.

Restarting the snap.multipass service ( `sudo snap restart multipass` ) and then trying `multipass start <name>` worked. And my K8s cluster recovered without issues. ;-)

**Comment by korikori** [comment]
I am seeing the same behavior as in the original reports - on server reboot, Multipass is unable to bring up the container and lists it as "Suspended". If I manually try to start it, the service is available in a browser, but the Multipass start command times out and then lists the container state as "Unknown". Running Multipass 1.8.0 on Ubuntu Server 21.10.

**Comment by gr1pcoding** [comment]
@townsend2010 Per your comment above,  wondering if you have any instructions on how I might recover a multipass instance. A multipass update to 1.11.1 on my M1 mac seems to have broken one I had created last year. I can create a new instance, but wanted to recover the one I've already got setup if possible. Thanks for your help!

FYI, error is:
start failed: The following errors occurred:
qemu-system-aarch64: Machine type received is 'virt-6.2' and local is 'virt-7.1'
qemu-system-aarch64: Error -22 while loading VM state
